### DISCUSSION TITLE
Отказ от NTP

### DISCUSSION THREAD

**Discussion by optenSTE** [Ar4w/wc_server#1] [Ideas]
Можно отказаться от использования NTP, заменив на встроенную функцию в FastBot. При отправке/приеме сообщения из него можно получить unix-время.
Так будет меньше зависимостей от внешней инфраструктуры.

**Comment by Ar4w** [comment]
Да, согласен. Когда буду проект здесь обновлять, то добавлю это изменение. 
Работу с флеш я уже переделал, но сюда пока не комитил. Хотел и энергонезависимую память на fastBot организовать но AlexGyver изменение одобрил но включит только в следующую версию
https://github.com/GyverLibs/FastBot/pull/59

**Reply by optenSTE** [reply]
Я, не зная ещё о вашем проекте, сделал аналогичный передатор показаний, во многом мы сошлись. От вашего добавлю очень полезный блок Мособлеирц (спасибо огромное за него).
Для сохранения текущих показаний счётчиков я использовал библиотеку Алекса https://github.com/GyverLibs/FileData
Сделал ограничение по частоте сохранения - каждые 10 тиков счётчика. У меня это 100л. Должно снять проблему с порчей ячеек памяти

**Reply by optenSTE** [reply]
Для получения информации о боте я использовал кастомное обращение к боту getMe https://tlgrm.ru/docs/bots/api
Возможно, там есть и нужное вам описание.

Кастомный запрос делал ещё на другой библиотеке, до FastBot

**Reply by Ar4w** [reply]
Я к стати уже переписал модуль хранения данных во флеш (только что закоммитил это изменение). Я взял всю не использующуюся область памяти

`eeprom,   data, nvs,     0x300000,0xF0000,
`
и поделил ее на три циклических массива

```
const uint32_t COL_OFF = 0x300000;
const uint32_t COL_LEN = 0x90000;
const uint32_t HOT_OFF = 0x390000;
const uint32_t HOT_LEN = 0x50000;
const uint32_t DATE_OFF = 0x3E0000;
const uint32_t DATE_LEN = 0x10000;

```
Размеры массивов пропорциональны частоте записи каждого параметра.
Так, я думаю, проблема с ресурсом Flash будет полностью снята

**Comment by optenSTE** [comment]
В своем проекте на FastBot я добавил загрузку прошивки ОТА. Всего несколько строк в обработчике входящих:

// обновить, если прошивку прислал Админ
  if (msg.OTA && cur_user_id == admin_user_ID)
  {
    send_bot_message("Обновление прошивки.\nНе отключайте питание!!!", chat_id);

    // сохранение в EEPROM
    initial_data_eeprom.updateNow();

    // перезагрузка
    bot.update();

    return;
  }
